python/pyarrow/public-api.pxi

```diff
@@ -96,6 +98,9 @@ cdef api object pyarrow_wrap_data_type(
         if ext_type.extension_name() == PyExtensionName():
             cpy_ext_type = <const CPyExtensionType*> ext_type
             return cpy_ext_type.GetInstance()
+        # elif ext_type.extension_name() == period_ext_name:
+        #     cgen_ext_type = <const CGenericExtensionType*> ext_type
+        #     return cgen_ext_type.GetInstance()
```

Comment:
For some reason, this doesn't work yet to get an actual instance back (it gives `None`). 
Also, I would need a way to check that the C type is actually a GenericExtensionType in the `elif` check.